src/hazmat/sieve.rs

```diff
@@ -140,14 +135,11 @@ impl<T: Integer> Sieve<T> {
         // Should not overflow since `incr` is never greater than `incr_limit`,
         // and the latter is chosen such that it doesn't overflow when added to `base`
         // (see the rest of this method).
-        self.base = self
-            .base
-            .checked_add(&self.incr.into())
-            .expect("Does not overflow by construction");
+        self.base = self.base.wrapping_add(&self.incr.into());
```

Comment:
Given that the code guaranteeing this contract is scattered around the module, it's probably better to keep `checked_add()`, just in case.